pkg/build/generator/generator.go

```diff
@@ -160,6 +160,31 @@ func describeBuildRequest(request *buildapi.BuildRequest) string {
 	return desc
 }
 
+// updateBuildEnv updates the strategy environment
+// This will replace the existing variable definitions with provided env
+func updateBuildEnv(strategy *buildapi.BuildStrategy, env []kapi.EnvVar) {
+	var buildEnv *[]kapi.EnvVar
+	switch strategy.Type {
+	case buildapi.SourceBuildStrategyType:
+		buildEnv = &strategy.SourceStrategy.Env
+	case buildapi.DockerBuildStrategyType:
+		buildEnv = &strategy.DockerStrategy.Env
+	case buildapi.CustomBuildStrategyType:
+		buildEnv = &strategy.CustomStrategy.Env
+	}
+	newEnv := []kapi.EnvVar{}
+	for _, e := range *buildEnv {
+		for _, n := range env {
+			if e.Name == n.Name {
+				continue
+			}
+			newEnv = append(newEnv, e)
```

Comment:
isn't this going to append "e" multiiple times, once for each "n" that doesn't match exactly?